components/merkledb/examples/migration.rs

```diff
@@ -0,0 +1,260 @@
+        let mut bytes = [0_u8; PUBLIC_KEY_LENGTH];
+        rng.fill(&mut bytes[..]);
+        let public_key = PublicKey::new(bytes);
+        let username = NAMES.choose(&mut rng).unwrap().to_string();
+        let wallet = v1::Wallet {
+            public_key,
+            username,
+            balance: rng.gen_range(0, 1_000),
+        };
+        schema.wallets.put(&public_key, wallet);
+
+        let history_len = rng.gen_range(0, 10);
+        schema
+            .histories
+            .get(&public_key)
+            .extend((0..history_len).map(|_| {
+                let mut bytes = [0_u8; HASH_SIZE];
+                rng.fill(&mut bytes[..]);
+                Hash::new(bytes)
+            }));
+    }
+}
+
+mod v2 {
+    use super::*;
+
+    #[derive(Debug, Serialize, Deserialize)]
+    pub struct Wallet {
+        pub username: String,
+        pub balance: u32,
+        pub history_hash: Hash, // << new field
+    }
+
+    impl BinaryValue for Wallet {
+        fn to_bytes(&self) -> Vec<u8> {
+            bincode::serialize(self).unwrap()
+        }
+
+        fn from_bytes(bytes: Cow<'_, [u8]>) -> Result<Self, Error> {
+            bincode::deserialize(bytes.as_ref()).map_err(From::from)
+        }
+    }
+
+    #[derive(Debug, Serialize, Deserialize)]
+    pub struct Config {
+        pub ticker: String,
+        pub divisibility: u8,
+    }
+
+    impl BinaryValue for Config {
+        fn to_bytes(&self) -> Vec<u8> {
+            bincode::serialize(self).unwrap()
+        }
+
+        fn from_bytes(bytes: Cow<'_, [u8]>) -> Result<Self, Error> {
+            bincode::deserialize(bytes.as_ref()).map_err(From::from)
+        }
+    }
+
+    impl_object_hash_for_binary_value! { Wallet, Config }
+
+    #[derive(Debug, FromAccess)]
+    pub struct Schema<T: Access> {
+        pub config: ProofEntry<T::Base, Config>,
+        pub wallets: ProofMapIndex<T::Base, PublicKey, Wallet>,
+        pub histories: Group<T, PublicKey, ProofListIndex<T::Base, Hash>>,
+    }
+
+    impl<T: Access> Schema<T> {
+        pub fn print_wallets(&self) {
+            for (public_key, wallet) in self.wallets.iter().take(10) {
+                println!("Wallet[{:?}] = {:?}", public_key, wallet);
+                println!(
+                    "History = {:?}",
+                    self.histories.get(&public_key).iter().collect::<Vec<_>>()
+                );
+            }
+        }
+    }
+}
+
+fn migrate(new_data: Prefixed<&Fork>, old_data: Prefixed<ReadonlyFork>) {
+    println!("\nStarted migration");
+    let old_schema = v1::Schema::new(old_data);
+    let mut new_schema = v2::Schema::new(new_data.clone());
+
+    // Move `ticker` and `divisibility` to `config`.
+    let config = v2::Config {
+        ticker: old_schema.ticker.get().unwrap(),
+        divisibility: old_schema.divisibility.get().unwrap_or(0),
+    };
+    new_schema.config.set(config);
+    // Mark these two indexes for removal.
+    new_data.clone().create_tombstone("ticker");
```

Comment:
Because `Prefixed` doesn't implement `Copy` and `create_tombstone` method consumes `self`, as all `AccessExt` methods do. Would it make more sense to have it consume `&self`, in your opinion?

Reply:
I think no need.